### ISSUE TITLE
TypeError: this.global.page.addListener is not a function

### ISSUE THREAD

**Issue by anooppv09** [puppeteer/puppeteer#5944]
I have facing this error in the new version 3.2.0
TypeError: this.global.page.addListener is not a function

**Comment by BryceMehring** [comment]
I'm getting this too within jest-puppeteer: https://github.com/smooth-code/jest-puppeteer/issues/353

Looks like `removeListener` was added for backwards compatibility but not `addListener`?

**Comment by rodoabad** [comment]
`3.2.0` should be a breaking change if we do not add backwards compatibility.

**Comment by mathiasbynens** [comment]
cc @jackfranklin related to #5907

**Comment by rodoabad** [comment]
Oh boy.

> Important this PR is a breaking change.

But we released a non-breaking change. Hahaha. Oops.

**Comment by SergeyFromHell** [comment]
I also has related issues with this change to new Mitt-based eventemitter.
I use 'newListener'/'removeListener' of a page object to remove dangling listeners on some error conditions.
[Node newListener event](https://nodejs.org/dist/latest-v14.x/docs/api/events.html#events_event_newlistener)

Could you please use another node-compatible implementation of eventemitter?
e.g. [events](https://www.npmjs.com/package/events)

**Comment by Tro95** [comment]
I'm experiencing this too:
`TypeError: page.removeAllListeners is not a function`

Is there an equivalent with `eventemitter.off()` for all listeners for the same event?

**Comment by jackfranklin** [comment]
Hi all,

Apologies for the issues here. We went back and forth between a major version or minor and (in hindsight, incorrectly) decided that we thought the events we added would make us backwards compatible enough which was obviously incorrect. We didn't think this behaviour was relied on greatly which it turns out it is.

Please stick to 3.1 for now and we'll figure out the best path forwards which will either be to revert back to EventEmitter or polyfill all the missing methods that we need to make this transition much smoother.

**Comment by Cuadrix** [comment]
@jackfranklin 
It also seems that removing a listener imported from another module doesn't actually remove the listener but just the reference of it, thus this;
```js
// someModule.js
// const page = .....
function myHandler(req) {
    // do something
}
function main(page) {
    page.off('request', myHandler);
}
module.exports = main;
```
```js
// main.js
const removeMyListener = require('./someModule')
// const page = .....
removeMyListener(page);
page.on('request', myHandler);
```
throws a 'Request is already handled!' error message when attempting to intercept requests after the fact.

The [mitt](https://github.com/developit/mitt/) that is used in `3.2.0` is too simple for this.

**Comment by AviVahl** [comment]
Chained listener registration also broke: `page.once('error', cb).once('pageerror', cb)`.

`once` no longer returns `this`, so the second `once` call breaks.

Could use auto-generated `.d.ts` to identify such issues at type-check time, now that the project's source is TypeScript-based.

**Comment by albatrossturkey1** [comment]
> Hi all,
> 
> Apologies for the issues here. We went back and forth between a major version or minor and (in hindsight, incorrectly) decided that we thought the events we added would make us backwards compatible enough which was obviously incorrect. We didn't think this behaviour was relied on greatly which it turns out it is.
> 
> Please stick to 3.1 for now and we'll figure out the best path forwards which will either be to revert back to EventEmitter or polyfill all the missing methods that we need to make this transition much smoother.

It broke the example in the readme.

**Comment by jackfranklin** [comment]
Thanks everyone for the extra info - sorry again for the issues. We're figuring out the best path forwards and will get something out soon that fixes the immediate problem and then we can plan a smoother migration forwards. Additionally we can improve our test coverage to make it more representative (one of the things that guided us was we didn't have a single test failure when we migrated the codebase) and catch these things ahead of time rather than place the burden on you all to report.

**Comment by jackfranklin** [comment]
Hi everyone, we have now released v3.3.0 which contains all the new features of v3.2.0 (the vision deficiency API for example) but **reverts the EventEmitter change**. We'll rethink how best to reintroduce that in a future version and ensure the process is much smoother.

If you have had problems with 3.2.0, please upgrade to 3.3.0 which does not have the problems that 3.2.0 had. 

Apologies again for the problems caused here. Please open a new issue if you're still having problems.
